src/index.ts

```diff
@@ -136,7 +136,7 @@ export function datadog<TEvent, TResult>(
         try {
           localResult = await promHandler(localEvent, localContext);
         } finally {
-          traceListener.onEndingInvocation(localEvent, localResult, finalConfig.captureLambdaPayload);
+          traceListener.onEndingInvocation(localEvent, localResult, metricsListener, localContext, finalConfig.captureLambdaPayload);
```

Comment:
Not sure if doing the logic in tracerListener is the best idea, but duplicating the logic from tracerListener to here is probably worse. Any suggestions are appreciated.

Reply:
What about : 
- making `onEndingInvocation` returning a boolean (true for success, false otherwise)
- call `incrementErrorsMetric(metricsListener, context);` in index (depending on the bool result)
This way you don't add coupling between trace and metric listeners?
WDYT?

Reply:
Sounds great! This approach is much cleaner.

Reply:
Alternatively you could add a new hook for the metrics listener called `onEndingInvocation` and pass the local result/context there. Either way makes sense I think.